formats/gltf/writer.go

```diff
@@ -519,23 +551,44 @@ func (w *Writer) AddTexture(polyTex PolyformTexture) *TextureInfo {
 		}
 	}
 
+	newTex := Texture{
+		Sampler: ptrI(samplerIndex),
+		Source:  ptrI(imageIndex),
+	}
+	newTexInfo := &TextureInfo{}
 	if len(texInfoExt) > 0 {
 		newTexInfo.Extensions = texInfoExt
 	}
 	if len(texExt) > 0 {
 		newTex.Extensions = texExt
 	}
 
-	w.textures = append(w.textures, newTex)
+	texIndex := len(w.textures)
+	var texFound bool
+texCompare:
```

Comment:
This is my first time seeing a goto in go, so I'm not understanding things as well as I should. Doesn't this cause a infinite loop if the extension data doesn't match?

Reply:
Gotcha! Thanks for the explanation

Reply:
Might be worth a refactor into an equals function if this logic is kept.

Reply:
Yeah, I'll look into it on Monday.